### DISCUSSION TITLE
JSLT Build

### DISCUSSION THREAD

**Discussion by hemakumarg89** [schibsted/jslt#201] [Q&A]
Perhaps a basic question. But I am bit struck here. pls help. TIA

I am trying to build the JSLT jar in my local using the gradle build. 
But  `./gradlew clean jar` command gives me following error.  Not sure why the code is not able to view the JSLTParser created by jjtree.  Any help is much appreciated. Thanks.

`> Task :core:jjtree
Java Compiler Compiler Version 4.1d1 (Tree Builder)
(type "jjtree" with no arguments for help)
Reading from file src/main/javacc/jslt.jj . . .
Warning: Output directory "/data/apps/hub/jslt/build/javacc/com/schibsted/spt/data/jslt/parser" does not exist. Creating the directory.
File "Node.java" does not exist.  Will create one.
File "SimpleNode.java" does not exist.  Will create one.
File "JsltParserTreeConstants.java" does not exist.  Will create one.
File "JJTJsltParserState.java" does not exist.  Will create one.
Annotated grammar generated successfully in /data/apps/hub/jslt/build/javacc/com/schibsted/spt/data/jslt/parser/jslt.jj.jj

> Task :core:javacc
Java Compiler Compiler Version 4.1d1 (Parser Generator)
(type "javacc" with no arguments for help)
Reading from file /data/apps/hub/jslt/build/javacc/com/schibsted/spt/data/jslt/parser/jslt.jj.jj . . .
Note: UNICODE_INPUT option is specified. Please make sure you create the parser/lexer using a Reader with the correct character encoding.
Warning: Choice conflict in (...)* construct at line 109, column 14.
         Expansion nested within construct and expansion following construct
         have common prefixes, one of which is: "|"
         Consider using a lookahead of 2 or more for nested expansion.
Warning: Choice conflict in [...] construct at line 115, column 13.
         Expansion nested within construct and expansion following construct
         have common prefixes, one of which is: "or"
         Consider using a lookahead of 2 or more for nested expansion.
Warning: Choice conflict in [...] construct at line 121, column 21.
         Expansion nested within construct and expansion following construct
         have common prefixes, one of which is: "and"
         Consider using a lookahead of 2 or more for nested expansion.
Warning: Choice conflict in [...] construct at line 127, column 18.
         Expansion nested within construct and expansion following construct
         have common prefixes, one of which is: "=="
         Consider using a lookahead of 2 or more for nested expansion.
Warning: Choice conflict in (...)* construct at line 148, column 24.
         Expansion nested within construct and expansion following construct
         have common prefixes, one of which is: "+"
         Consider using a lookahead of 2 or more for nested expansion.
Warning: Choice conflict in (...)* construct at line 161, column 14.
         Expansion nested within construct and expansion following construct
         have common prefixes, one of which is: "*"
         Consider using a lookahead of 2 or more for nested expansion.
Warning: Choice conflict in [...] construct at line 185, column 40.
         Expansion nested within construct and expansion following construct
         have common prefixes, one of which is: <IDENT>
         Consider using a lookahead of 2 or more for nested expansion.
Warning: Choice conflict in [...] construct at line 186, column 3.
         Expansion nested within construct and expansion following construct
         have common prefixes, one of which is: "."
         Consider using a lookahead of 2 or more for nested expansion.
Warning: Choice conflict in [...] construct at line 193, column 3.
         Expansion nested within construct and expansion following construct
         have common prefixes, one of which is: "."
         Consider using a lookahead of 2 or more for nested expansion.
Warning: Choice conflict in [...] construct at line 285, column 5.
         Expansion nested within construct and expansion following construct
         have common prefixes, one of which is: "else"
         Consider using a lookahead of 2 or more for nested expansion.
File "TokenMgrError.java" does not exist.  Will create one.
File "ParseException.java" does not exist.  Will create one.
File "Token.java" does not exist.  Will create one.
File "SimpleCharStream.java" does not exist.  Will create one.
Parser generated with 0 errors and 10 warnings.

> Task :core:compileJava FAILED
/data/apps/hub/jslt/core/src/main/java/com/schibsted/spt/data/jslt/parser/ParserImpl.java:52: error: cannot find symbol
  public static Expression compileExpression(ParseContext ctx, JsltParser parser) {
                                                               ^
  symbol:   class JsltParser
  location: class ParserImpl
/data/apps/hub/jslt/core/src/main/java/com/schibsted/spt/data/jslt/parser/ParserImpl.java:80: error: cannot find symbol
  private static ExpressionImpl compileModule(ParseContext ctx, JsltParser parser) {
                                                                ^
  symbol:   class JsltParser
  location: class ParserImpl
/data/apps/hub/jslt/core/src/main/java/com/schibsted/spt/data/jslt/parser/ParserImpl.java:93: error: cannot find symbol
  private static ExpressionImpl compile(ParseContext ctx, SimpleNode root) {
                                                          ^
  ...
  ...
  ...
                       ^
  symbol:   variable JsltParserConstants
  location: class ParserImpl
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
100 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':core:compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/4.10.3/userguide/command_line_interface.html#sec:command_line_warnings
`

**Comment by larsga** [comment]
The `JsltParser` class is created by javacc. You may want to check if the file is actually there:

```
Larss-MacBook-Pro-2:jslt larsga$ ll core/build/javacc/com/schibsted/spt/data/jslt/parser/
total 328
-rw-r--r--  1 larsga  staff   3378 Jul  1 15:22 JJTJsltParserState.java
-rw-r--r--  1 larsga  staff  54599 Jul  1 15:22 JsltParser.java
-rw-r--r--  1 larsga  staff   3074 Jul  1 15:22 JsltParserConstants.java
-rw-r--r--  1 larsga  staff  27423 Jul  1 15:22 JsltParserTokenManager.java
-rw-r--r--  1 larsga  staff   1903 Jul  1 15:22 JsltParserTreeConstants.java
-rw-r--r--  1 larsga  staff   1289 Jul  1 15:22 Node.java
-rw-r--r--  1 larsga  staff   6692 Jul  1 15:22 ParseException.java
-rw-r--r--  1 larsga  staff  12102 Jul  1 15:22 SimpleCharStream.java
-rw-r--r--  1 larsga  staff   2495 Jul  1 15:22 SimpleNode.java
-rw-r--r--  1 larsga  staff   3841 Jul  1 15:22 Token.java
-rw-r--r--  1 larsga  staff   4474 Jul  1 15:22 TokenMgrError.java
-rw-r--r--  1 larsga  staff  25874 Jul  1 15:22 jslt.jj.jj
```

That should at least tell you whether the problem is that the file isn't created, or whether it's there but can't be found.

From your output I can see the javacc task is at least invoked.
